src/Raven.Server/Documents/Handlers/ReplicationHandler.cs

```diff
@@ -41,6 +43,60 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
                 });
             }
         }
+        
+        [RavenAction("/databases/*/debug/replication/all-items", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+        public async Task GetAllItems()
+        {
+            var etag = GetLongQueryString("etag", required: false) ?? 0L;
+            var pageSize = GetPageSize();
+
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            using (context.OpenReadTransaction())
+            {
+                var runStats = new OutgoingReplicationRunStats();
+                var stats = new ReplicationDocumentSender.ReplicationStats
+                {
+                    Network = new OutgoingReplicationStatsScope(runStats),
+                    Storage = new OutgoingReplicationStatsScope(runStats),
+                    AttachmentRead = new OutgoingReplicationStatsScope(runStats),
+                    CounterRead = new OutgoingReplicationStatsScope(runStats),
+                    DocumentRead = new OutgoingReplicationStatsScope(runStats),
+                    TombstoneRead = new OutgoingReplicationStatsScope(runStats),
+                    TimeSeriesRead = new OutgoingReplicationStatsScope(runStats),
+                };
+                var array = new DynamicJsonArray();
+
+                foreach (ReplicationBatchItem item in ReplicationDocumentSender.GetReplicationItems(Database, context, etag, stats, false).Take(pageSize))
+                {
+                    var djv = new DynamicJsonValue
+                    {
+                        [nameof(item.ChangeVector)] = item.ChangeVector,
+                        [nameof(item.Etag)] = item.Etag,
+                        [nameof(item.Type)] = item.Type,
+                        [nameof(item.TransactionMarker)] = item.TransactionMarker,
+                    };
+                    switch (item)
+                    {
+                        case DocumentReplicationItem i:
+                            djv[nameof(i.Collection)] = i.Collection;
+                            djv[nameof(i.Id)] = i.Id;
+                            djv[nameof(i.Flags)] = i.Flags;
+                            break;
+                        case AttachmentReplicationItem a:
+                            djv[nameof(a.Base64Hash)] = a.Base64Hash.ToString();
+                            djv[nameof(a.Name)] = a.Name.ToString();
+                            djv[nameof(a.ContentType)] = a.ContentType.ToString();
+                            break;
+                    }
```

Comment:
this is handy, let's maybe add `ToJson` to the `ReplicationBatchItem`

Reply:
That is a bigger change that we have to do right now. 
And a virtual method call is likely slower than this FWIW